### PR TITLE
Feature:add Jaeger to provide distributed tracing

### PR DESCRIPTION
**Pull request by lyt122**
To finish the issue https://github.com/CocaineCong/tangseng/issues/54. This uses a tracer for the go part of tangseng with the opentelemetry.

### CODE REVIEW

File: pkg/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+)
+
+func InitTracerProvider(Url string, serviceName string) func(ctx context.Context) error {
```
Comment (CocaineCong) [inline pkg/tracing/tracing.go:36]: can u add some code comments ?

File: app/gateway/routes/router.go [hunk 2]
```diff
@@ -18,18 +18,20 @@
 package routes
 
 import (
+	"github.com/CocaineCong/tangseng/consts"
```
Comment (CocaineCong) [inline app/gateway/routes/router.go:21]: this is a problem with the sort of importing packages obviously

**Review by CocaineCong — COMMENTED** [review]
can u traceing the mysql or redis in this way ? cause it is important that we should know the cost of time in mysql or redis in the request.

**Comment by CocaineCong** [comment]
BTW, same the problem with #https://github.com/CocaineCong/tangseng/pull/26

**Comment by CocaineCong** [comment]
have u tested the code ? or can u post the result or something else about your test ?

**Comment by lyt122** [comment]
> have u tested the code ? or can u post the result or something else about your test ?

when you run the project and make a request, you can go to localhost:16686 to see the results.

**Comment by CocaineCong** [comment]
> > have u tested the code ? or can u post the result or something else about your test ?
> 
> when you run the project and make a request, you can go to localhost:16686 to see the results.

I mean where is trace id ?

**Comment by lyt122** [comment]
> > > have u tested the code ? or can u post the result or something else about your test ?
> > 
> > 
> > when you run the project and make a request, you can go to localhost:16686 to see the results.
> 
> I mean where is trace id ?

trace id is in the context.
when a grpc/http request started, a span will be generated，this span has a trace id

**Comment by CocaineCong** [comment]
> > > > have u tested the code ? or can u post the result or something else about your test ?
> > > 
> > > 
> > > when you run the project and make a request, you can go to localhost:16686 to see the results.
> > 
> > 
> > I mean where is trace id ?
> 
> trace id is in the context. when a grpc/http request started, a span will be generated，this span has a trace id

why no return trace id in response body ? that is reason why I say the same problem with https://github.com/CocaineCong/tangseng/pull/26

**Review by CocaineCong — APPROVED** [review]
LGTM

**Pull request merged**